jwst/guider_cds/tests/test_guider_cds.py

```diff
@@ -0,0 +1,143 @@
+            exp_type == model.meta.exposure.type)
+
+
+def test_guider_cds_fineguide_mode(make_guider_image):
+    """Test the fine guiding mode
+    """
+
+    model = make_guider_image
+    model.meta.exposure.type = 'FGS_FINEGUIDE'
+
+    truth = np.zeros(model.data.shape)
+
+    result = guider_cds(model)
+
+    n_int = model.data.shape[0]
+    imshape = (model.data.shape[2], model.data.shape[3])
+
+    slope_int_cube = np.zeros((n_int,) + imshape, dtype=np.float32)
+
+    for num_int in range(0, n_int):
+        data_sect = model.data[num_int, :, :, :]
+        first_4 = data_sect[:4, :, :].mean(axis=0)
+        last_4 = data_sect[-4:, :, :].mean(axis=0)
+        slope_int_cube[num_int, :, :] = last_4 - first_4
+
+        truth = slope_int_cube /  model.meta.exposure.group_time
+
+    assert np.array_equal(result.data, truth)
+
+
+@pytest.mark.parametrize("exptype", ['FGS_ACQ1', 'FGS_ACQ2', 'FGS_TRACK'])
+def test_guider_cds_acq_track_modes(exptype, make_guider_image):
+    """Test acq and track exptypes"""
+
+    model = make_guider_image
+    model.meta.exposure.type = exptype
+
+    truth = np.zeros(model.data.shape)
+
+    result = guider_cds(model)
+
+    n_int = model.data.shape[0]
+    imshape = (model.data.shape[2], model.data.shape[3])
+    slope_int_cube = np.zeros((n_int,) + imshape, dtype=np.float32)
+
+    for num_int in range(0, n_int):
+        data_sect = model.data[num_int, :, :, :]
+        grp_last = data_sect[1, :, :]
+        grp_first = data_sect[0, :, :]
+        slope_int_cube[num_int, :, :] = grp_last - grp_first
+
+        truth = slope_int_cube /  model.meta.exposure.group_time
+
+    assert np.array_equal(result.data, truth)
+
+@pytest.mark.parametrize("exptype", ['FGS_ID-IMAGE', 'FGS_ID-STACK'])
+def test_guider_cds_id_modes(exptype, make_guider_image):
+    model = make_guider_image
+    model.meta.exposure.type = exptype
+
+    result = guider_cds(model)
+
+    n_int = model.data.shape[0]
+    imshape = (model.data.shape[2], model.data.shape[3])
+
+    truth = np.zeros((1,) + imshape)
+
+    for num_int in range(0, n_int):
+        data_sect = model.data[num_int, :, :, :]
+        grp_last = data_sect[1, :, :]
+        grp_first = data_sect[0, :, :]
+
+        if num_int == 0:
+            diff_int0 = grp_last - grp_first
+        if num_int == 1:
+            diff_int1 = grp_last - grp_first
+
+    truth[0, :, :] = np.minimum(diff_int1, diff_int0) / model.meta.exposure.group_time
+
+    assert np.array_equal(result.data[0,:,:], truth[0,:,:])
+
+def test_unit_assignment(make_guider_image):
+    """Test that correct units are returned"""
+
+    model = make_guider_image
+    model.meta.exposure.type = "FGS_FINEGUIDE"
+
+    result = guider_cds(model)
+
+    assert result.meta.bunit_data == 'DN/s'
+
+def test_table_extensions(make_guider_image):
+    """Test that tables are assigned to result of pipeline"""
+
+    model = make_guider_image
+
+    model.planned_star_table = np.arange(0, 11)
```

Comment:
This is interesting; I'm surprised that it works.  These are tables, so I would expect the schema verification to fail.  I'm not really concerned, as long as it continues to work.

Reply:
I see what you mean, I can change it now if you'd prefer it to have the same shape as the data.